### ISSUE TITLE
Bad state: No element

### ISSUE THREAD

**Issue by kevinxxq** [bosskmk/pluto_grid#331]
Hi,
I got an error said **Bad state: No element**.

My use case:
```
PlutoGrid(
        key: UniqueKey(),
        onLoaded: (PlutoGridOnLoadedEvent event) {
          controller.stateManager = event.stateManager;
        },
        configuration: getPlutoGridConfiguration(context),
        mode: PlutoGridMode.select,
        columns: controller.listCols,
        rows: controller.listRows,
      ),
```
I don't know why plutogrid can't set columns from controller.listCols which is built earlier than it(At this time, controller.listRows does not contain element).

The stacktrace of this error:
When the exception was thrown, this was the stack: 
#0      Iterable.first (dart:core/iterable.dart:490:7)
#1      ColumnState.currentColumn (package:pluto_grid/src/manager/state/column_state.dart:193:10)
#2      EditingState.autoEditing (package:pluto_grid/src/manager/state/editing_state.dart:52:23)
#3      CellState.setCurrentCell (package:pluto_grid/src/manager/state/cell_state.dart:207:16)
#4      _PlutoGridState.initSelectMode.<anonymous closure> (package:pluto_grid/src/pluto_grid.dart:253:22)
#5      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1145:15)
#6      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1090:9)
#7      SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:996:5)
#11     _invoke (dart:ui/hooks.dart:150:10)
#12     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:270:5)
#13     _drawFrame (dart:ui/hooks.dart:114:31)

I check the source code:

```
PlutoColumn? get currentColumn {
    if (currentColumnField == null) {
      return null;
    }

    return refColumns!
        .where((element) => element.field == currentColumnField)
        .first;
  }
```

From the source:
dart:core PlutoColumn get first  Containing class: Iterable  
Returns the first element.
Throws a StateError if this is empty. Otherwise returns the first element in the iteration order, equivalent to this.elementAt(0).
  Dart Packages

I suggest replace .first with .firstWhereOrNull() but I am not sure whether it will hide some cases need prompt the exception.

Thank you.

**Comment by bosskmk** [comment]
First of all, thank you for your report.

The situation where currentColumnField is not null and not in refColumns is because the column is added dynamically?
There is currently no ability to dynamically add columns, but they will be added.
We will update to take this into consideration.

thank you.

**Comment by bosskmk** [comment]
Related to #307

**Comment by kevinxxq** [comment]
> 

Thank you. 
Columns are generated in advance and not added dynamically.

It is my problem, I don't know the relationship between stateManager.Cols and controller.listCols(if I set cols:controller.listCols).

The above error takes place when I set rows: controller.listRows and controller.listRows contains 0 elements. 
If I set rows:[], it works without problem, but the plutogrid can't keep the data during switching tab pages, I wrap the tab page with plutogrid inside a keepalive widget to solve this.

Now it works without problem. Expect your update for dynamic columns, at that time, this problem may be eliminated together.

Thank you.
